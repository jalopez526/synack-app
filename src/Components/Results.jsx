import React from "react";
import Result from "./Result";

const Results = ({ data }) => {
  if (!data) return <></>;

  return (
    <div data-testid="result" className="item-container">
      {data.map((result, index) => (
        <Result
          key={index}
          title={result.title}
          link={result.link}
          secondary={result.snippet}
        />
      ))}
    </div>
  );
};

export default Results;
