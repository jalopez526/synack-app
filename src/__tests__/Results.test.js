import Results from "../Components/Results";
import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';

describe("Testing Results component", () => {
  const data = [
    {
      title: "Google",
      link: "www.google.com",
      snippet: "Google Inc.",
    },
    {
      title: "Bing",
      link: "www.bing.com",
      snippet: "Bing Inc.",
    },
  ];

  it('renders correctly', () => {
    const tree = renderer
      .create(<Results data={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it("should render items from Results", () => {
    const { getByTestId } = render(<Results data={data} />);
    expect(getByTestId("result")).toBeInTheDocument();
  });

  it("should render no items from Results", () => {
    const { queryByTestId } = render(<Results />);
    expect(queryByTestId("result")).toBeFalsy();
  });
});
