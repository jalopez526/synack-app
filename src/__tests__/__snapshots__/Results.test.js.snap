// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Results component renders correctly 1`] = `
<div
  className="item-container"
  data-testid="result"
>
  <p
    data-testid="item"
  >
    <a
      href="www.google.com"
      rel="noreferrer"
      target="_blank"
    >
      Google
    </a>
     
    <br />
    <small>
      Google Inc.
    </small>
  </p>
  <hr
    className="hr"
  />
  <p
    data-testid="item"
  >
    <a
      href="www.bing.com"
      rel="noreferrer"
      target="_blank"
    >
      Bing
    </a>
     
    <br />
    <small>
      Bing Inc.
    </small>
  </p>
  <hr
    className="hr"
  />
</div>
`;
